File: java/src/test/java/codingdojo/WordWrapTest.java
package codingdojo;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;


public class WordWrapTest {

    static Stream<WordWrap> wrapperProvider() {
        return Stream.of(
                new WordWrapKoflerLoop(),
                new WordWrapKoflerLoopBuffer(),
                new WordWrapKoflerRegex(),
                new WordWrapBlausten(),
                new WordWrapBlausten2(),
                new WordWrapMartin()
        );
    }

    // TODO: write some more test cases like this one
    @ParameterizedTest
    @MethodSource("wrapperProvider")
    public void testWrapEmptyString(WordWrap wrapper) {
        assertEquals("", wrapper.wrap("", 1));
    }

}

File: typescript/src/WordWrap.test.ts
import {WordWrapBlausten} from "./WordWrapBlausten";
import {WordWrapBlausten2} from "./WordWrapBlausten2";
import {WordWrapKoflerLoop} from "./WordWrapKoflerLoop";
import {WordWrapKoflerRegex} from "./WordWrapKoflerRegex";
import {WordWrapMartin} from "./WordWrapMartin";
import WordWrap from "./WordWrap";

describe.each([
    ["WordWrapBlausten", new WordWrapBlausten()],
    ["WordWrapBlausten2", new WordWrapBlausten2()],
    ["WordWrapKoflerLoop", new WordWrapKoflerLoop()],
    ["WordWrapKoflerRegex", new WordWrapKoflerRegex()],
    ["WordWrapMartin", new WordWrapMartin()]
])("WordWrap implemented with %s", (name, wrapper: WordWrap) => {

    // TODO: write some more test cases like this one
    it('wraps an empty string ', () => {
        expect(wrapper.wrap("", 1)).toBe("");
    });
})